File: Chap8-Internationalization/8-1InternationalizationDemo/settingsdialog.cpp
#include "settingsdialog.h"
#include "ui_settingsdialog.h"
#include <QDebug>
#include <QSettings>

SettingsDialog::SettingsDialog(QWidget *parent) :
    QDialog(parent),
    ui(new Ui::SettingsDialog)
{
    ui->setupUi(this);
}

SettingsDialog::~SettingsDialog()
{
    delete ui;
}

void SettingsDialog::on_buttonBox_accepted()
{
    qDebug() << "Pressed OK";
    QSettings settings("Blikoon Technologies", "Painter App");

    switch (ui->languageCombobox->currentIndex()) {

    case 0 : {
        //Default
        qDebug() << "Active language is default";
        settings.setValue("language","default");

        break;
    }
    case 1 : {
        //English
        qDebug() << "Active language is english";
        settings.setValue("language","english");

        break;
    }
    case 2 : {
        //French
        qDebug() << "Active language is french";
        settings.setValue("language","french");

        break;
    }
    case 3 : {
        //Chinese
        qDebug() << "Active language is chinese";
        settings.setValue("language","chinese");

        break;
    }

    }
}

void SettingsDialog::on_buttonBox_rejected()
{
    qDebug() << "Pressed Cancel";
}

File: Chap8-Internationalization/8-1InternationalizationDemo/settingsdialog.py
from PySide6.QtWidgets import QDialog
from PySide6.QtCore import QSettings
import ui_settingsdialog  # We'll need to generate this with pyside6-uic

class SettingsDialog(QDialog):
    """Dialog for changing language settings"""
    
    def __init__(self, parent=None):
        super().__init__(parent)
        self.ui = ui_settingsdialog.Ui_SettingsDialog()
        self.ui.setupUi(self)
    
    def accept(self):
        """Handle OK button click"""
        settings = QSettings("Blikoon Technologies", "Painter App")
        
        # Save the selected language
        index = self.ui.languageCombobox.currentIndex()
        if index == 0:
            # Default
            settings.setValue("language", "default")
        elif index == 1:
            # English
            settings.setValue("language", "english")
        elif index == 2:
            # French
            settings.setValue("language", "french")
        elif index == 3:
            # Chinese
            settings.setValue("language", "chinese")
        
        super().accept()